Untitled Pull Request

Heepsound opened the pull request:
Выполнено задание по sprint_7

Heepsound committed:
Выполнено задание по sprint_4

Heepsound committed:
Изменено шрифт кнопок на YP/Medium/20

Heepsound committed:
Изменено команда разработчика

Heepsound committed:
Выполнено задание по sprint_5

Heepsound committed:
Выполнено задание по sprint_6

Heepsound committed:
Убраны пустые строки в конце классов.
Для всех @IBOutlet добавил weak и private.
Добавил final для для классов AlertPresenter и QuestionFactory

Heepsound committed:
Выполнено задание по sprint_7

Helvetii951 commented on MovieQuiz/Models/GameRecord.swift:
```diff
@@ -0,0 +1,19 @@
+    let total: Int
+    let date: Date
+    
+    func isBetterThan(_ another: GameRecord) -> Bool {
+        correct > another.correct
+    }
```
Часто для сравнения сущностей реализуют протокол [Comparable](https://developer.apple.com/documentation/swift/comparable/).

Helvetii951 commented on MovieQuiz/Presentation/AlertPresenter.swift:
```diff
@@ -0,0 +1,35 @@
+    
+    // MARK: - Private functions
+    
+    func show(_ alertModel: AlertModel) {
+        let alert = UIAlertController(title: alertModel.title,
+                                      message: alertModel.message,
+                                      preferredStyle: .alert)
+        alert.view.accessibilityIdentifier = "Results"
```
Функцию можно сделать статической, в качестве еще одно параметра передавать контроллер. Так на не нужно будет создавать экземпляр `AlertPresenter`. Создание экземпляров классов - не самая дешевая операция.

Helvetii951 commented on MovieQuiz/Services/QuestionFactory.swift:
```diff
@@ -0,0 +1,112 @@
+            DispatchQueue.main.async {
+                guard let self = self else { return }
+                switch result {
+                case .success(let mostPopularMovies):
+                    self.movies = mostPopularMovies.items // сохраняем фильм в нашу новую переменную
+                    self.delegate?.didLoadDataFromServer() // сообщаем, что данные загрузились
+                case .failure(let error):
+                    self.delegate?.didFailToLoadData(with: error) // сообщаем об ошибке нашему MovieQuizViewController
+                }
```
Swift - достаточно выразительный язык. Часто можно понять что происходит и без комментариев.

Helvetii951 commented on MovieQuiz/Services/QuestionFactory.swift:
```diff
@@ -0,0 +1,112 @@
+        }
+    }
+    
+    /*
+    private let questions: [QuizQuestion] = [
+        QuizQuestion(
+            image: "The Godfather",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: true),
+        QuizQuestion(
+            image: "The Dark Knight",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: true),
+        QuizQuestion(
+            image: "Kill Bill",
```
Это последний спринт. Лучше очистить проект от ненужных комментариев.

Heepsound committed:
Реализовано сравнение сущностей "GameRecord" через протокол Codable.
Отображение алерта реализована через статическую функцию, а не через создание экземпляра класса.
Удалены лишние комментарии в коде.

Heepsound committed:
Добавлено тест скрытия алерта после нажатия на кнопку на нём (проверить что алерт отсутствует, счётчик установлен на 1/10)
Время ожидания изменено с 2 на 3 секунды

Heepsound committed:
Добавил паузы в 3 секунды в начале тестов
